src/tile-board/Tileboard.jsx

```diff
@@ -2,15 +2,27 @@ import React, { useState } from "react";
 import Tile from "./Tile";
 
 function TileBoard() {
-  const [arrayGrid] = useState(() => createArrayGrid());
+  const [size] = useState(5);
+  const [arrayGrid, setArrayGrid] = useState(() => createArrayGrid(size));
+
+  const handleToggle = ({ target }) => {
+    setArrayGrid(toggle(arrayGrid, target.id, size));
+  };
 
   return (
     <div className="container text-center mt-5">
       {arrayGrid.map((row, keyi) => {
         return (
           <div key={`${keyi}`}>
             {row.map((tile, keyj) => {
-              return <Tile key={`${keyi}, ${keyj}`} />;
+              return (
+                <Tile
+                  key={`${keyi}${keyj}`}
+                  id={`${keyi}${keyj}`}
+                  isOn={tile}
+                  handleToggle={handleToggle}
```

Comment:
Rather than relying on the `target.id` in your callback, I would just pass through the x/y indices directly here:

```js
handleToggle={() => handleToggle(keyi, keyj)
```

I might also name those loop variables `x` and `y` for clarity.

This also avoids a bug in your `toggle` function if either of these variables is more than 1 digit since currently you're indexing into the string id at position 0 and 1 which would break if one of the indexes were e.g. "12".